ros2/src/geoscenario_server/geoscenario_server/geoscenario_server.py

```diff
@@ -0,0 +1,258 @@
+            if self.local_cartesian_projector is None:
+                self.local_cartesian_projector = LocalCartesianProjector(
+                        Origin(origin["origin_lat"], origin["origin_lon"], origin["origin_alt"])
+                    )
+
+        for vehicle in vehicles:
+            msg = Vehicle()
+            msg.id = vehicle["id"]
+            msg.type = vehicle["type"]
+            msg.dimensions.x = vehicle["l"]
+            msg.dimensions.y = vehicle["w"]
+            msg.dimensions.z = vehicle["h"]
+            self.set_msg_pos_vel_from_agent(msg, vehicle)
+            msg.yaw = vehicle["yaw"]
+            msg.steering_angle = vehicle["steering_angle"]
+            tick_msg.vehicles.append(msg)
+
+        for pedestrian in pedestrians:
+            msg = Pedestrian()
+            msg.id = pedestrian["id"]
+            msg.type = pedestrian["type"]
+            msg.dimensions.x = pedestrian["l"]
+            msg.dimensions.y = pedestrian["w"]
+            msg.dimensions.z = pedestrian["h"]
+            self.set_msg_pos_vel_from_agent(msg, pedestrian)
+            msg.yaw = pedestrian["yaw"]
+            tick_msg.pedestrians.append(msg)
+
+        self.tick_pub.publish(tick_msg)
+        self.previous_tick_count = tick_count
+
+
+    def tick_from_client(self, msg):
+        # convert the msg into dictionaries
+        # ignore origin, simulation_time, type, yaw, steering angle not used in the client
+        vehicles = []
+        for msg_vehicle in msg.vehicles:
+            vehicle = {}
+            vehicle["id"] = msg_vehicle.id
+            vehicle["type"] = msg_vehicle.type # not used
+            vehicle["l"] = msg_vehicle.dimensions.x
+            vehicle["w"] = msg_vehicle.dimensions.y
+            vehicle["h"] = msg_vehicle.dimensions.z
+            self.set_agent_pos_vel_from_msg(vehicle, msg_vehicle)
+            vehicle["yaw"] = msg_vehicle.yaw # not used
+            vehicle["steering_angle"] = msg_vehicle.steering_angle # not used
+            vehicle["active"] = msg_vehicle.active
+            vehicles.append(vehicle)
+
+        pedestrians = []
+        for msg_pedestrian in msg.pedestrians:
+            pedestrian = {}
+            pedestrian["id"] = msg_pedestrian.id
+            pedestrian["type"] = msg_pedestrian.type # not used
+            pedestrian["l"] = msg_pedestrian.dimensions.x
+            pedestrian["w"] = msg_pedestrian.dimensions.y
+            pedestrian["h"] = msg_pedestrian.dimensions.z
+            self.set_agent_pos_vel_from_msg(pedestrian, msg_pedestrian)
+            pedestrian["yaw"] = msg_pedestrian.yaw # not used
+            pedestrian["active"] = msg_vehicle.active
+            pedestrians.append(pedestrian)
+
+        # Write to shared memory for dashboard and traffic
+        # Vehicle states will be updated by SimTraffic.tick() reading from CS shared memory
+        self.sim_client_shm.write_client_state(msg.tick_count, msg.delta_time, vehicles, pedestrians)
```

Comment:
this is actually the shared memory for co-simulators (client/server). We don't want to use this here.
That is what `geoscenario_client` used to communicate with the standalone GSServer.